realm/realm-library/src/main/java/io/realm/internal/modules/CompositeMediator.java

```diff
@@ -106,6 +106,9 @@ public void insert(Realm realm, RealmModel object, Map<RealmModel, Long> cache)
 
     @Override
     public void insert(Realm realm, Collection<? extends RealmModel> objects) {
+        if (objects.isEmpty()) {
```

Comment:
Hm, empty collection is a valid input. Shouldn't every *Mediator be able to handle all valid input correctly?